v2/spiffetls/dial.go

```diff
@@ -59,9 +59,9 @@ func DialWithMode(ctx context.Context, network, addr string, mode DialMode, opti
 	case tlsClientMode:
 		tlsconfig.HookTLSClientConfig(tlsConfig, m.bundle, m.authorizer)
```

Comment:
We should fix this to pass the options once `HookTLSClientConfig` has been patched to accept them (see my comment in tlsconfig/config.go).